lib/content-services/src/lib/search/components/search-chip-input/search-chip-input.component.ts

```diff
@@ -0,0 +1,72 @@
+/*!
+ * @license
+ * Copyright © 2005-2023 Hyland Software, Inc. and its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ENTER } from '@angular/cdk/keycodes';
+import { Component, EventEmitter, Input, OnInit, Output, ViewEncapsulation } from '@angular/core';
+import { MatChipInputEvent } from '@angular/material/chips';
+import { Observable } from 'rxjs';
+
+@Component({
+    selector: 'adf-search-chip-input',
+    templateUrl: './search-chip-input.component.html',
+    styleUrls: ['./search-chip-input.component.scss'],
+    encapsulation: ViewEncapsulation.None,
+    host: { class: 'adf-search-chip-input' }
+})
+export class SearchChipInputComponent implements OnInit {
+    @Input()
+    label: string;
+
+    @Input()
+    addOnBlur = true;
+
+    @Input()
+    onReset: Observable<void>;
+
+    @Output()
+    phrasesChanged: EventEmitter<string[]> = new EventEmitter();
+
+    phrases: string[] = [];
+    readonly separatorKeysCodes = [ENTER] as const;
+
+    ngOnInit() {
+        this.onReset?.subscribe(() => this.resetChips());
+    }
+
+    addPhrase(event: MatChipInputEvent) {
+        const phrase = (event.value || '').trim();
+
+        if (phrase) {
+          this.phrases.push(phrase);
+          this.phrasesChanged.emit(this.phrases);
+        }
+        event.chipInput.clear();
+    }
+
+    removePhrase(phrase: string) {
+        const index = this.phrases.indexOf(phrase);
```

Comment:
I think the element that calls this method is already looping through the phrases array through ngFor. Maybe you can track the indices in that loop, and then pass that index as a parameter to this method? It would save you from having to do the indexOf() call in here. You would have to test it out though

Reply:
Good point, changed